### PR TITLE
[WIP] OSDOCS#3995: ROSA - port 'Security' content from OCP

### PR DESCRIPTION
**Pull request by tmalove**
This PR is to port the "Security and compliance" OCP content to ROSA. Reference that section only for comments, reviews, etc.
[OSDOCS-3995](https://issues.redhat.com//browse/OSDOCS-3995)

Version(s): 

Link to docs preview (local build): https://file.rdu.redhat.com/tlove/sd-port-security-tlove/security/index.html (Updated 3/5)

QE review:
- [ ] QE has approved this change.

### CODE REVIEW

**Comment by tmalove** [comment]
@jaybeeunix  See this spreadsheet of errors I get from testing CLI commands in the security section. This list is still WIP. Thx.
[https://docs.google.com/document/d/1EE4o5iF9bXMZS20DeeuJkd-UakZnKssVXcjsaFrPvx4/edit#heading=h.g1x54zpbotj](url)

**Comment by xiaojiey** [comment]
@xingxingxia @geliu2016 @sunilcio @wangke19 Could you please help to review for your components? thanks.
I will check for Compliance Operator, File Integrity Operator and Security Profiles Operator.

File: _topic_maps/_topic_map_rosa.yml [hunk 1]
```diff
@@ -237,6 +237,220 @@ Topics:
...
+  - Name: Authenticating the cert-manager Operator with AWS Security Token Service
+    File: cert-manager-authenticate-aws
+  - Name: Configuring log levels for cert-manager and the cert-manager Operator for Red Hat OpenShift
+    File: cert-manager-log-levels
+  - Name: Authenticating the cert-manager Operator for Red Hat OpenShift with GCP Workload Identity
+    File: cert-manager-authenticate-gcp
+  - Name: Authenticating the cert-manager Operator for Red Hat OpenShift on AWS
+    File: cert-manager-authentication-non-sts
+  - Name: Authenticating the cert-manager Operator for Red Hat OpenShift on GCP
+    File: cert-manager-authenticate-non-sts-gcp
```
Comment (xingxingxia) [inline _topic_maps/_topic_map_rosa.yml:421]: It is ROSA document for ROSA cluster, then GCP documents `cert-manager-authenticate-gcp` and `cert-manager-authenticate-non-sts-gcp` are not applicable, need be removed.
`cert-manager-operator-issuer-acme` has GCP ambient credential section, that is also not applicable for ROSA cluster.

**Comment by BhargaviGudi** [comment]
For file-integrity-operator, generally it is good. There are minor changes need to be updated:
1. Commands under below links needs to be appended with namespace name (-n openshift-file-integrity)
a. https://file.rdu.redhat.com/tlove/sd-port-security-tlove/security/file_integrity_operator/file-integrity-operator-understanding.html#checking-the-file-integrity-CR-status_file-integrity-operator
b. https://file.rdu.redhat.com/tlove/sd-port-security-tlove/security/file_integrity_operator/file-integrity-operator-understanding.html#understanding-file-integrity-node-statuses-object_file-integrity-operator
c. https://file.rdu.redhat.com/tlove/sd-port-security-tlove/security/file_integrity_operator/file-integrity-operator-understanding.html#file-integrity-node-status-failure_file-integrity-operator
d. https://file.rdu.redhat.com/tlove/sd-port-security-tlove/security/file_integrity_operator/file-integrity-operator-understanding.html#file-integrity-events_file-integrity-operator
e. https://file.rdu.redhat.com/tlove/sd-port-security-tlove/security/file_integrity_operator/file-integrity-operator-configuring.html#file-integrity-examine-default-config_file-integrity-operator
f. https://file.rdu.redhat.com/tlove/sd-port-security-tlove/security/file_integrity_operator/file-integrity-operator-configuring.html#file-integrity-operator-defining-custom-config_file-integrity-operator
g. https://file.rdu.redhat.com/tlove/sd-port-security-tlove/security/file_integrity_operator/file-integrity-operator-advanced-usage.html#file-integrity-operator-reinitializing-database_file-integrity-operator
h. https://file.rdu.redhat.com/tlove/sd-port-security-tlove/security/file_integrity_operator/file-integrity-operator-troubleshooting.html#determining-the-fileintegrity-objects-phase

2. For Understanding the FileIntegrityNodeStatuses object, https://file.rdu.redhat.com/tlove/sd-port-security-tlove/security/file_integrity_operator/file-integrity-operator-understanding.html#understanding-file-integrity-node-statuses-object_file-integrity-operator , _Example Output_ needs to be updated.
Sample output:
<pre>
$ oc get fileintegritynodestatuses -n openshift-file-integrity
NAME                                                              NODE                                         STATUS
worker-fileintegrity-ip-10-0-145-177.us-east-2.compute.internal   ip-10-0-145-177.us-east-2.compute.internal   Succeeded
worker-fileintegrity-ip-10-0-171-252.us-east-2.compute.internal   ip-10-0-171-252.us-east-2.compute.internal   Succeeded
worker-fileintegrity-ip-10-0-193-251.us-east-2.compute.internal   ip-10-0-193-251.us-east-2.compute.internal   Succeeded
</pre>
3. Defining a custom File Integrity Operator configuration,https://file.rdu.redhat.com/tlove/sd-port-security-tlove/security/file_integrity_operator/file-integrity-operator-configuring.html#file-integrity-operator-defining-custom-config_file-integrity-operator , Step #7 need to update command to create _fileintegrity_. 
Currently only fileintegrity config is given.
Add command: `$ oc create -f master-fileintegrity.yaml -n openshift-file-integrity`

**Comment by tmalove** [comment]
> Apiserver topics look good to me. But I checked the link [file.rdu.redhat.com/tlove/sd-port-security-tlove/upgrading/rosa-upgrading-cluster-prepare.html](https://file.rdu.redhat.com/tlove/sd-port-security-tlove/upgrading/rosa-upgrading-cluster-prepare.html), we still has this upgrade path, I think that's a problem.
> 
> ```
> Preparing to upgrade ROSA to 4.9 
> ```

@wangke19 When this was initially added, the admin acknowledgment was a new requirement. However, all major versions require this now. I will follow up with SRE to confirm. Thanks.

**Comment by tmalove** [comment]
@sheriff-rh @bergerhoffer @xenolinux  adding you for your awareness.

**Review by bergerhoffer — COMMENTED** [review]
Just one minor thing from what I can tell (looking at the impact to the OCP docs). Thanks!

**Comment by sheriff-rh** [comment]
Looks good, Tami! Nice work with those ifdefs :)

**Comment by BhargaviGudi** [comment]
> For Compliance Operator, below points are not applicable:
> 
> 1. For important notice in[ https://file.rdu.redhat.com/tlove/sd-port-security-tlove/security/compliance_operator/compliance-operator-supported-profiles.html](https://file.rdu.redhat.com/tlove/sd-port-security-tlove/security/compliance_operator/compliance-operator-supported-profiles.html), better to remove unreleated platforms:
>    Current: "The Compliance Operator might report incorrect results on managed platforms, such as OpenShift Dedicated, Red Hat OpenShift Service on AWS, and Azure Red Hat OpenShift. For more information, see the[ Red Hat Knowledgebase Solution #6983418](https://access.redhat.com/solutions/6983418)."
>    Better to be: "The Compliance Operator might report incorrect results on Red Hat OpenShift Service on AWS. For more information, see the[ Red Hat Knowledgebase Solution #6983418](https://access.redhat.com/solutions/6983418)."
> 2. For important notice in[ https://file.rdu.redhat.com/tlove/sd-port-security-tlove/security/compliance_operator/compliance-operator-installation.html](https://file.rdu.redhat.com/tlove/sd-port-security-tlove/security/compliance_operator/compliance-operator-installation.html),
>    Current: "The Compliance Operator might report incorrect results on managed platforms, such as OpenShift Dedicated, Red Hat OpenShift Service on AWS, and Microsoft Azure Red Hat OpenShift. For more information, see the[ Red Hat Knowledgebase Solution #6983418](https://access.redhat.com/solutions/6983418)."
>    Better to be: "The Compliance Operator might report incorrect results on Red Hat OpenShift Service on AWS. For more information, see the[ Red Hat Knowledgebase Solution #6983418](https://access.redhat.com/solutions/6983418)."
> 3. For paragraph below, it is not applicable for ROSA as it is for hypershift hosted cluster only
>    https://file.rdu.redhat.com/tlove/sd-port-security-tlove/security/compliance_operator/compliance-operator-installation.html#installing-compliance-operator-hcp_compliance-operator-installation
> 4. For paragraph below, it is not applicable for ROSA as creating custom mcp will fail. https://docs.openshift.com/container-platform/4.13/security/compliance_operator/compliance-operator-remediation.html#compliance-operator-apply-remediation-for-customized-mcp
>    https://docs.openshift.com/container-platform/4.13/security/compliance_operator/compliance-operator-remediation.html#compliance-custom-node-pools_compliance-remediation

@tmalove We could see that 1st and 2nd points are resolved as expected. But 3rd and 4th points have not resolved yet. Could you please help me check? Thanks

**Comment by BhargaviGudi** [comment]
> Add one more important point for Compliance Operator: The auto-remediation not works for ROSA(more accurately for all management platforms). Please add important notice in doc. That means below sections not work for ROSA: https://docs.openshift.com/container-platform/4.13/security/compliance_operator/compliance-operator-remediation.html#compliance-applying_compliance-remediation https://docs.openshift.com/container-platform/4.13/security/compliance_operator/compliance-operator-remediation.html#compliance-manual_compliance-remediation https://docs.openshift.com/container-platform/4.13/security/compliance_operator/compliance-operator-remediation.html#compliance-updating_compliance-remediation https://docs.openshift.com/container-platform/4.13/security/compliance_operator/compliance-operator-remediation.html#compliance-unapplying_compliance-remediation

Need to add important notice for Compliance Operator as auto-remediation doesn't work for ROSA.

**Comment by tmalove** [comment]
@xiaojiey @xingxingxia @geliu2016 @sunilcio @wangke19 Except for the item about the 'Upgrading to 4.9', the requested changes are made and the preview link is updated. This PR is for the Security and compliance section only. I still need to verify the 4.9 reference and will let you know. Thanks!

**Comment by BhargaviGudi** [comment]
File-integrity operator and Security-profile operator looks good.
Note: There is a known documentation bug for SPO [OCPBUGS-18377](https://issues.redhat.com/browse/OCPBUGS-18377)

**Comment by sheriff-rh** [comment]
Hi @BhargaviGudi, I'll look into the SPO bug. Thanks for filing it!

**Comment by xiaojiey** [comment]
1. For the installation for compliance operator, only this "[Installing the Compliance Operator on Hosted control planes](https://docs.openshift.com/container-platform/4.13/security/compliance_operator/compliance-operator-installation.html#installing-compliance-operator-hcp_compliance-operator-installation)" should be removed as it is for hypershift hosted cluster only. The other [installation part](https://docs.openshift.com/container-platform/4.13/security/compliance_operator/compliance-operator-installation.html) should be kept.
2. One more issue in https://62384--docspreview.netlify.app/openshift-rosa/latest/security/compliance_operator/compliance-scans, for the notice  "For all-in-one control plane and worker nodes, the compliance scan  runs twice on the worker and control plane nodes. The compliance scan  might generate inconsistent scan results. You can avoid inconsistent  results by defining only a single role in the ScanSetting object", I am not sure if it is applicable for ROSA. I will double check and update next Monday.

**Comment by xiaojiey** [comment]
> 1. For the installation for compliance operator, only this "[Installing the Compliance Operator on Hosted control planes](https://docs.openshift.com/container-platform/4.13/security/compliance_operator/compliance-operator-installation.html#installing-compliance-operator-hcp_compliance-operator-installation)" should be removed as it is for hypershift hosted cluster only. The other part [installation part](https://docs.openshift.com/container-platform/4.13/security/compliance_operator/compliance-operator-installation.html) should be kept.
> 
>     2. One more issue in https://62384--docspreview.netlify.app/openshift-rosa/latest/security/compliance_operator/compliance-scans, for the notice  "For all-in-one control plane and worker nodes, the compliance scan  runs twice on the worker and control plane nodes. The compliance scan  might generate inconsistent scan results. You can avoid inconsistent  results by defining only a single role in the ScanSetting object", I am not sure if it is applicable for ROSA. I will double check and update next Monday.
Confirmed that ROSA don't have all-in-one configuration. I think it not applicable to ROSA.

File: modules/checking-file-intergrity-cr-status.adoc [hunk 2]
```diff
@@ -12,10 +12,19 @@ The `FileIntegrity` custom resource (CR) reports its status through the .`status
 
 * To query the `FileIntegrity` CR status, run:
 +
+ifndef::openshift-dedicated,openshift-rosa[]
 [source,terminal]
 ----
-$ oc get fileintegrities/worker-fileintegrity  -o jsonpath="{ .status.phase }"
+$ oc get fileintegrities/worker-fileintegrity  -o jsonpath="{ .status.phase }"
```
Comment (xiaojiey) [inline modules/checking-file-intergrity-cr-status.adoc:18]: Please remove the extra space at the end of the line

**Comment by tmalove** [comment]
@xiaojiey @xingxingxia @geliu2016 @sunilcio @wangke19  The sections under the 'Compliance Operator' has been reorganized into subsections. Letting you know this for awareness. I am verifying the latest link against your comments here. At your earliest will you let me know if there is anything that you want addressed? Thanks!

**Comment by tmalove** [comment]
@wangke19  I am in the process of verifying that we can change the content for preparing to upgrade a ROSA cluster from 4.9 only to include later versions.

**Comment by tmalove** [comment]
@karanthshashank Will you assist in getting the final acks and I will get this into peer review. Thanks!

**Comment by xiaojiey** [comment]
@tmalove Thanks a lot for all the update.
One more important point:
For Compliance Operator, in page https://file.rdu.redhat.com/tlove/sd-port-security-tlove/security/compliance_operator/co-scans/compliance-operator-remediation.html， the auto-remediation could not be applied on ROSA(more accurately for all management platforms). Please add important notice in doc. 
For example, below description is wrong as ROSA admnistrator cannot appy the auto remediation.
`Unless requested, the remediations are not applied automatically, which gives an Red Hat OpenShift Service on AWS administrator the opportunity to review what the remediation does and only apply a remediation once it has been verified.`
cc: @rhmdnd @sheriff-rh Correct me if I am wrong. Thanks.

**Comment by sheriff-rh** [comment]
That is my understanding, as well - but I defer to @rhmdnd to verify.

**Comment by tmalove** [comment]
@xiaojiey  The preview is updated for this section. Let me know your thoughts. Thanks. 

image

**Comment by xiaojiey** [comment]
Looks good to me now.

**Comment by rhmdnd** [comment]
Looks good so long as we keep something in the documentation about potentially reporting incorrect results on managed offerings (at least for the time being until we land support for supporting managed offerings in the profiles).

**Comment by arendej** [comment]
Based on looking at the https://62384--docspreview.netlify.app/openshift-rosa/latest there is still much to be adjusted for ROSA specifics.

**Comment by tmalove** [comment]
@xiaojiey @xingxingxia @geliu2016 @sunilcio @wangke19  Have these sections been verified for ROSA?
 - Compliance Operator
 - File Integrity Operator
Also, for 'Configuring certificates' and its subsections, is this possible for ROSA? Should we mention that this task is managed by SRE?

Thanks!

**Comment by tmalove** [comment]
> Based on looking at the [62384--docspreview.netlify.app/openshift-rosa/latest](https://62384--docspreview.netlify.app/openshift-rosa/latest) there is still much to be adjusted for ROSA specifics.

Hi @arendej I am working with PM to make some specific changes and also with QE.

**Comment by xiaojiey** [comment]
@tmalove Really sorry for one 

> @xiaojiey @xingxingxia @geliu2016 @sunilcio @wangke19 Have these sections been verified for ROSA?
> 
> * Compliance Operator
> * File Integrity Operator
>   Also, for 'Configuring certificates' and its subsections, is this possible for ROSA? Should we mention that this task is managed by SRE?
> 
> Thanks!

For Compliance Operator, yes.
And really sorry for one more comment, can we add this part ["Installing the Compliance Operator on Hosted control planes"](https://docs.openshift.com/container-platform/4.14/security/compliance_operator/co-management/compliance-operator-installation.html#installing-compliance-operator-hcp_compliance-operator-installation) back? I forgot the ROSA hosted cp config in previous comments, and now realized is necessary for the hosted clusters of ROSA hosted cp.

**Comment by xiaojiey** [comment]
> @xiaojiey In reference to your PR, [openshift/openshift-tests-private#12742](https://github.com/openshift/openshift-tests-private/pull/12742), what information from this is needed in the docs? Thanks!
No. The background of the PR is that for standalone OCP, we have a hardcoded catalogsource called qe-app-registry. However, it is not available on the managed clusters. So I applied a patch that when qe-app-registry not available, test the released version(with catalogsource redhat-operators) instead. Otherwise the test won't be executed on the managed clusters.

**Comment by BhargaviGudi** [comment]
@tmalove File integrity operator and compliance operator looks good to me.

**Comment by tmalove** [comment]
Hi @arendej Will you verify this PR? QE has already acked it. Thanks!

**Comment by arendej** [comment]
still not ready. addressed in an internal slack thread

**Comment by tmalove** [comment]
@xiaojiey Will you have someone from QE assigned to review this PR? Thanks!

**Comment by tmalove** [comment]
> still not ready. addressed in an internal slack thread

Hi @arendej  What channel is this discussed in? Thx.

**Comment by xiaojiey** [comment]
For Compliance Operator, the release notes for [Compliance-operatorv1.4.0](https://docs.openshift.com/container-platform/4.14/security/compliance_operator/compliance-operator-release-notes.html#compliance-operator-release-notes-1-4-0) is available. All other looks good to me.

File: _topic_maps/_topic_map_rosa.yml [hunk 3]
```diff
@@ -811,9 +1048,8 @@ Topics:
 - Name: Deployments
   Dir: deployments
   Distros: openshift-rosa
-  Topics:
```
Comment (aireilly) [inline _topic_maps/_topic_map_rosa.yml:814]: Put back this line. removing it breaks the build

Comment (aireilly) [reply _topic_maps/_topic_map_rosa.yml:814]: Whole stanza should look like: 

```
Topics:
- Name: Deployments
  Dir: deployments
  Distros: openshift-rosa
  Topics:
  - Name: Custom domains for applications
    File: osd-config-custom-domains-applications
 ```

**Comment by tmalove** [comment]
@arendej @rhmdnd @BillDett the preview is updated with the removal of the  'Compliance Operator' and 'File Integrity Operator' sections FYI. Thanks!

**Comment by codymant** [comment]
@tmalove the netflix preview link doesn't seem to work any more. Is there another link somewhere? Eric Chapman and Austin Quam would like to review and share feedback if the review window is still open.

https://62384--ocpdocs-pr.netlify.app/openshift-rosa/latest/welcome/index.html

**Comment by tmalove** [comment]
> @tmalove the netflix preview link doesn't seem to work any more. Is there another link somewhere? Eric Chapman and Austin Quam would like to review and share feedback if the review window is still open.
> 
> [62384--ocpdocs-pr.netlify.app/openshift-rosa/latest/welcome/index.html](https://62384--ocpdocs-pr.netlify.app/openshift-rosa/latest/welcome/index.html)

@codymant An updated preview is available, however VPN/network access is required. Let me know if Eric or Austin cannot access it.

**Comment by tmalove** [comment]
This PR is superseded by https://github.com/openshift/openshift-docs/pull/72837. Refer to this PR for the Security porting project.

**Comment by xingxingxia** [comment]
> @rhmdnd @BillDett @arendej @xingxingxia Use this latest https://github.com/openshift/openshift-docs/pull/72837 for updates on the 'Security and compliance' porting.

@tmalove got it. For me, what I reviewed was for cert-manager as in my previous comment https://github.com/openshift/openshift-docs/pull/62384#discussion_r1289681301 . In the new PR, seems cert-manager doc is totally removed. Anyway nvm, I'm transfering to @lunarwhite to continue review for cert-manager area in the new PR.

**Pull request closed without merging**